File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if(list1==null){
            return list2;
        }
        if(list2==null){
            return list1;
        }
        ListNode ans = new ListNode(-1);
        ListNode curr = ans;
        while(list1!=null && list2!=null){
            if(list1.val<=list2.val){
                curr.next = list1;
                list1 = list1.next;
            }
            else if(list2.val<=list1.val){
                curr.next = list2;
                list2 = list2.next;
                
            }
            curr = curr.next;            
        }
        curr.next = list1==null?list2:list1;
        return ans.next;
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        l1 = list1
        l2 = list2
        ans = ListNode(0)
        head = ans
        while l1 and l2:
            if l1.val <= l2.val:
                ans.next = l1
                l1 = l1.next
            elif l1.val > l2.val:
                ans.next = l2
                l2 = l2.next
            ans = ans.next
        while l1:
            ans.next = l1
            l1 = l1.next
            ans = ans.next
        while l2:
            ans.next = l2
            l2 = l2.next
            ans = ans.next
        return head.next